boa/environment.py

```diff
@@ -341,6 +344,14 @@ def gas_meter_class(self, cls):
     def enable_gas_profiling(self) -> None:
         self.set_gas_meter_class(ProfilingGasMeter)
 
+    @contextlib.contextmanager
+    def store_call_profile(self, enable: bool = True) -> None:
+        try:
+            self.profile_calls = enable
+            yield
+        finally:
+            self.profile_calls = False
+
```

Comment:
I'm not convinced that we need to expose this on the environment object. I can't think of why a user would want to use it -- it's pretty much only there for testing purposes at this time. instead of having this method, we should just inline this code in `pytest_runtest_call`

Reply:
Well they could use this for their own statistics. This just offers a way to capture gas used. in boa.env.profiled_calls dict. Nothing more really.